app-games/src/main/java/cm/aptoide/pt/app_games/appview/AppViewScreen.kt

```diff
@@ -0,0 +1,313 @@
+  navigateBack: () -> Unit,
+  navigate: (String) -> Unit,
+) = animatedComposable(
+  appViewRoute
+) { it ->
+  val packageName = it.arguments?.getString("packageName")!!
+  AppViewScreen(
+    packageName = packageName,
+    navigateBack = navigateBack,
+    navigate = { navigate(it) },
+  )
+}
+
+fun buildAppViewRoute(
+  packageName: String,
+): String = "app/$packageName"
+
+@Composable
+fun AppViewScreen(
+  packageName: String = "",
+  navigateBack: () -> Unit = {},
+  navigate: (String) -> Unit = {},
+) {
+  val appViewModel = appViewModel(packageName = packageName, adListId = "")
+  val uiState by appViewModel.uiState.collectAsState()
+
+  MainAppViewView(
+    uiState = uiState,
+    reload = appViewModel::reload,
+    noNetworkReload = {
+      appViewModel.reload()
+    },
+    navigateBack = {
+      navigateBack()
+    },
+    tabsList = tabsList
+  )
+}
+
+@Composable
+fun MainAppViewView(
+  uiState: AppUiState,
+  reload: () -> Unit,
+  noNetworkReload: () -> Unit,
+  navigateBack: () -> Unit,
+  tabsList: List<AppViewTab>,
+) {
+  when (uiState) {
+    is AppUiState.Idle ->
+      AppViewContent(
+        app = uiState.app,
+        tabsList = tabsList,
+        navigateBack = navigateBack,
+      )
+
+    is AppUiState.NoConnection -> NoConnectionView(onRetryClick = noNetworkReload)
+    is AppUiState.Error -> GenericErrorView(reload)
+    is AppUiState.Loading -> LoadingView()
+  }
+}
+
+@Composable
+fun LoadingView() {
+  Column(
+    modifier = Modifier
+      .fillMaxWidth()
+      .fillMaxHeight(),
+    horizontalAlignment = Alignment.CenterHorizontally,
+    verticalArrangement = Arrangement.Center
+  ) {
+    CircularProgressIndicator()
+  }
+}
+
+@Composable
+fun AppViewContent(
+  app: App,
+  tabsList: List<AppViewTab>,
+  navigateBack: () -> Unit,
+) {
+  val selectedTab by rememberSaveable { mutableIntStateOf(0) }
+  val appImageString = stringResource(id = string.app_view_image_description_body, app.name)
+
+  val scrollState = rememberScrollState()
+
+  Column(
+    modifier = Modifier.verticalScroll(scrollState)
+  ) {
+    Box {
+      AptoideFeatureGraphicImage(
+        modifier = Modifier
+          .clearAndSetSemantics { contentDescription = appImageString }
+          .graphicsLayer {
+            translationY = scrollState.value * 0.8f
+          }
+          .height(FEATURE_GRAPHIC_HEIGHT.dp)
+          .fillMaxWidth(),
+        data = app.featureGraphic,
+        contentDescription = null
+      )
+      Image(
+        imageVector = AppTheme.icons.LeftArrow,
+        contentDescription = stringResource(id = string.button_back_title),
+        contentScale = ContentScale.Crop,
+        modifier = Modifier
+          .clickable(onClick = navigateBack)
+          .padding(horizontal = 16.dp, vertical = 12.dp)
+          .size(32.dp)
+      )
+    }
+    Column(
+      modifier = Modifier.background(AppTheme.colors.background)
+    ) {
+
+      AppPresentationView(app)
+
+      AppGamesButton(
```

Comment:
This button is missing the negative vertical offset. Not very important, but this would be a design issue